web/models.py

```
@@ -1308,6 +1308,67 @@ def display_name(self):
+
+    # New fields added as per feedback
+    created_at = models.DateTimeField(auto_now_add=True)
+    modified_at = models.DateTimeField(auto_now=True)
+
+    def __str__(self):
+        course_title = self.course.title if self.course else "No Course"
+        return f"Certificate for {self.user.username} - {course_title}"
+
+    def clean(self):
+        """Validate that the user has completed the course."""
+        from django.core.exceptions import ValidationError
```

Ruff (0.8.2)

`django.core.exceptions.ValidationError` imported but unused

Remove unused import: `django.core.exceptions.ValidationError`

(F401)